// vocs/_lib/vite/.vocs/dist/assets/getting-started-Ca-J-Kce.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
import { G as GenerateKey } from "./GenerateKey-DHuufDZQ.js";
import "react";
import "lucide-react";
import "clsx";
const frontmatter = {
  "title": "Get started",
  "description": "Get started with Whisk SDK in just a few lines of code."
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    li: "li",
    p: "p",
    pre: "pre",
    span: "span",
    ul: "ul",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsxs(_components.header, {
      children: [jsxs(_components.h1, {
        id: "get-started",
        children: ["Get started", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#get-started",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsx(_components.div, {
        role: "doc-subtitle",
        children: "Get started with Whisk SDK in just a few lines of code."
      })]
    }), "\n", jsxs(_components.h2, {
      id: "overview",
      children: ["Overview", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#overview",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Whisk SDK provides a suite of modular kits for seamless integration with the ", jsx(_components.a, {
        href: "https://whisk.so",
        children: "Whisk.so"
      }), " blockchain data pipeline.\nEach kit includes:"]
    }), "\n", jsxs(_components.ul, {
      children: ["\n", jsx(_components.li, {
        children: "Customizable React components for rapid development."
      }), "\n", jsx(_components.li, {
        children: "React Hooks programmatic access to Whisk data."
      }), "\n", jsx(_components.li, {
        children: "Direct graphQL access for low level control and use outside of React."
      }), "\n"]
    }), "\n", jsx(_components.p, {
      children: jsx(_components.img, {
        src: "/how-it-works.png",
        alt: "Whisk Diagram"
      })
    }), "\n", jsxs(_components.h2, {
      id: "installation",
      children: ["Installation", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#installation",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.div, {
      className: "code-group",
      children: [jsx(_components.div, {
        "data-title": "npm",
        children: jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            "data-title": "npm",
            "data-lang": "bash",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "npm"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " i"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " @paperclip-labs/whisk-sdk"
                })]
              })
            })
          })
        })
      }), jsx(_components.div, {
        "data-title": "pnpm",
        children: jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            "data-title": "pnpm",
            "data-lang": "bash",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "pnpm"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " i"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " @paperclip-labs/whisk-sdk"
                })]
              })
            })
          })
        })
      }), jsx(_components.div, {
        "data-title": "yarn",
        children: jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            "data-title": "yarn",
            "data-lang": "bash",
            children: jsx(_components.code, {
              children: jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "yarn"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " add"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: " @paperclip-labs/whisk-sdk"
                })]
              })
            })
          })
        })
      })]
    }), "\n", jsxs(_components.h2, {
      id: "configuration",
      children: ["Configuration", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#configuration",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.div, {
      "data-vocs-steps": "true",
      children: [jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "import-styles-at-the-top-of-your-layout-or-app",
          children: ["Import styles at the top of your layout or app", jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#import-styles-at-the-top-of-your-layout-or-app",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            "data-title": "layout.tsx",
            "data-lang": "ts",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "import"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: ' "@paperclip-labs/whisk-sdk/styles.css"'
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "// Other style imports below"
                })
              })]
            })
          })
        })]
      }), jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "generate-api-key",
          children: ["Generate API key", jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#generate-api-key",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsx(_components.p, {
          children: "The API key will be sent to your email address."
        }), jsx(GenerateKey, {})]
      }), jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "wrap-app-in-whisksdkprovider",
          children: ["Wrap App in ", jsx(_components.code, {
            children: "WhiskSdkProvider"
          }), jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#wrap-app-in-whisksdkprovider",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsxs(_components.div, {
          className: "code-group",
          children: [jsx(_components.div, {
            "data-title": "providers.tsx",
            children: jsx(Fragment, {
              children: jsx(_components.pre, {
                className: "shiki shiki-themes github-light github-dark-dimmed has-focused twoslash lsp",
                style: {
                  backgroundColor: "#fff",
                  "--shiki-dark-bg": "#22272e",
                  color: "#24292e",
                  "--shiki-dark": "#adbac7"
                },
                tabIndex: "0",
                "data-title": "providers.tsx",
                "data-lang": "tsx",
                children: jsxs(_components.code, {
                  children: [jsxs(_components.span, {
                    className: "line",
                    children: [jsxs(_components.span, {
                      style: {
                        color: "#032F62",
                        "--shiki-dark": "#96D0FF"
                      },
                      children: ['"', "use", " ", "client", '"']
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: ";"
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "import"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: " { "
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "function"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#DCBDFB"
                              },
                              children: " WhiskSdkProvider"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "({ "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "apiKey"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "config"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: " }"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " WhiskSdkProviderParams"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ")"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " JSX"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "Element"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "WhiskSdkProvider"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: " } "
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "from"
                    }), jsxs(_components.span, {
                      style: {
                        color: "#032F62",
                        "--shiki-dark": "#96D0FF"
                      },
                      children: [' "@', "paperclip", "-", "labs", "/", "whisk", "-", "sdk", '"']
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "; "
                    })]
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    "data-empty-line": true,
                    children: " "
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "export"
                    }), jsxs(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: [" ", "default"]
                    }), jsxs(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: [" ", "function"]
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#DCBDFB"
                      },
                      children: " "
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#DCBDFB"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsxs(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: [jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "function"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#DCBDFB"
                              },
                              children: " Providers"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "({ "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: " }"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: " {"
                            })]
                          }), "\n", jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "    children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "ReactNode"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ";"
                            })]
                          }), "\n", jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "})"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "JSX"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "Element"
                            })]
                          })]
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "Providers"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "({ "
                    }), jsx(_components.span, {
                      style: {
                        color: "#E36209",
                        "--shiki-dark": "#F69D50"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ": React.ReactNode"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "children"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " }"
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: ":"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " { "
                    }), jsx(_components.span, {
                      style: {
                        color: "#E36209",
                        "--shiki-dark": "#F69D50"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ": React.ReactNode"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "children"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: ":"
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " "
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#F69D50"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "export"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " namespace"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "React"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "."
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#F69D50"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsxs(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: [jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "type"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ".ReactNode "
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "="
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: " string"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " |"
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: " number"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " |"
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: " boolean"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " |"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "ReactElement"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "<"
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: "any"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: "string"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " |"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "JSXElementConstructor"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "<"
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: "any"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ">> "
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "|"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " Iterable"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "<"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "ReactNode"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "> "
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "|"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " React"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "ReactPortal"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " |"
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: " null"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: " |"
                            }), jsx(_components.span, {
                              style: {
                                color: "#005CC5",
                                "--shiki-dark": "#6CB6FF"
                              },
                              children: " undefined"
                            })]
                          }), jsxs(_components.div, {
                            className: "twoslash-popup-jsdoc",
                            children: [jsx(_components.p, {
                              children: "Represents all of the things React can render."
                            }), "\n", jsxs(_components.p, {
                              children: ["Where ReactElement only represents JSX, ", jsx(_components.code, {
                                children: "ReactNode"
                              }), " represents everything that can be rendered."]
                            })]
                          })]
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "ReactNode"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " }) "
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "{"
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsxs(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: ["  ", "return"]
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: " ("
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "    <"
                    }), jsx(_components.span, {
                      style: {
                        color: "#005CC5",
                        "--shiki-dark": "#8DDB8C"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "function"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#DCBDFB"
                              },
                              children: " WhiskSdkProvider"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "({ "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "apiKey"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "config"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: " }"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " WhiskSdkProviderParams"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ")"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " JSX"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "Element"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "WhiskSdkProvider"
                        })]
                      })
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#6CB6FF"
                      },
                      children: "        "
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#6CB6FF"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsx(_components.span, {
                            className: "line",
                            children: jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "WhiskSdkProviderParams.apiKey: string"
                            })
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "apiKey"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "="
                    }), jsxs(_components.span, {
                      style: {
                        color: "#032F62",
                        "--shiki-dark": "#96D0FF"
                      },
                      children: ['"<', "YOUR_API_KEY", '>"']
                    }), jsx(_components.span, {
                      style: {
                        color: "#6A737D",
                        "--shiki-dark": "#768390"
                      },
                      children: " // Check your email from prev step. "
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#6CB6FF"
                      },
                      children: "        "
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#6CB6FF"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsx(_components.span, {
                            className: "line",
                            children: jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "WhiskSdkProviderParams.config: WhiskSdkConfig"
                            })
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "config"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "="
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F47067"
                      },
                      children: "{"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "{ "
                    })]
                  }), "\n", jsx(_components.span, {
                    className: "line focused",
                    children: jsx(_components.span, {
                      style: {
                        color: "#6A737D",
                        "--shiki-dark": "#768390"
                      },
                      children: "            // Define configuration for each kit you plan to use.  "
                    })
                  }), "\n", jsx(_components.span, {
                    className: "line focused",
                    children: jsx(_components.span, {
                      style: {
                        color: "#6A737D",
                        "--shiki-dark": "#768390"
                      },
                      children: "            // See respective Kit docs for full config. "
                    })
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "        }"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F47067"
                      },
                      children: "}"
                    })]
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    children: jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "    >"
                    })
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F47067"
                      },
                      children: "        {"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ": React.ReactNode"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "children"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F47067"
                      },
                      children: "}"
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "    </"
                    }), jsx(_components.span, {
                      style: {
                        color: "#005CC5",
                        "--shiki-dark": "#8DDB8C"
                      },
                      children: jsxs(_components.span, {
                        className: "twoslash-hover",
                        children: [jsx(_components.div, {
                          className: "twoslash-popup-info-hover",
                          children: jsxs(_components.span, {
                            className: "line",
                            children: [jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: "function"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#DCBDFB"
                              },
                              children: " WhiskSdkProvider"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "({ "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "apiKey"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "config"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ", "
                            }), jsx(_components.span, {
                              style: {
                                color: "#E36209",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "children"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: " }"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " WhiskSdkProviderParams"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: ")"
                            }), jsx(_components.span, {
                              style: {
                                color: "#D73A49",
                                "--shiki-dark": "#F47067"
                              },
                              children: ":"
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: " JSX"
                            }), jsx(_components.span, {
                              style: {
                                color: "#24292E",
                                "--shiki-dark": "#ADBAC7"
                              },
                              children: "."
                            }), jsx(_components.span, {
                              style: {
                                color: "#6F42C1",
                                "--shiki-dark": "#F69D50"
                              },
                              children: "Element"
                            })]
                          })
                        }), jsx(_components.span, {
                          className: "twoslash-target",
                          children: "WhiskSdkProvider"
                        })]
                      })
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "> "
                    })]
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    children: jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "  ); "
                    })
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    children: jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "}"
                    })
                  })]
                })
              })
            })
          }), jsx(_components.div, {
            "data-title": "layout.tsx",
            children: jsx(Fragment, {
              children: jsx(_components.pre, {
                className: "shiki shiki-themes github-light github-dark-dimmed has-focused",
                style: {
                  backgroundColor: "#fff",
                  "--shiki-dark-bg": "#22272e",
                  color: "#24292e",
                  "--shiki-dark": "#adbac7"
                },
                tabIndex: "0",
                "data-title": "layout.tsx",
                "data-lang": "tsx",
                children: jsxs(_components.code, {
                  children: [jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "import"
                    }), jsx(_components.span, {
                      style: {
                        color: "#032F62",
                        "--shiki-dark": "#96D0FF"
                      },
                      children: ' "@paperclip-labs/whisk-sdk/styles.css"'
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "import"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: " { Providers } "
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "from"
                    }), jsx(_components.span, {
                      style: {
                        color: "#032F62",
                        "--shiki-dark": "#96D0FF"
                      },
                      children: ' "./providers"'
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "; "
                    })]
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    "data-empty-line": true,
                    children: " "
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "export"
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: " default"
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: " function"
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#DCBDFB"
                      },
                      children: " RootLayout"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "({ "
                    }), jsx(_components.span, {
                      style: {
                        color: "#E36209",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "children"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " }"
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: ":"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " { "
                    }), jsx(_components.span, {
                      style: {
                        color: "#E36209",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "children"
                    }), jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: ":"
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " React"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "."
                    }), jsx(_components.span, {
                      style: {
                        color: "#6F42C1",
                        "--shiki-dark": "#F69D50"
                      },
                      children: "ReactNode"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F69D50"
                      },
                      children: " }) "
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "{"
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#D73A49",
                        "--shiki-dark": "#F47067"
                      },
                      children: "  return"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: " ("
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "    <"
                    }), jsx(_components.span, {
                      style: {
                        color: "#005CC5",
                        "--shiki-dark": "#8DDB8C"
                      },
                      children: "Providers"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "> // [!code focus]"
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F47067"
                      },
                      children: "        {"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "children"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#F47067"
                      },
                      children: "}"
                    })]
                  }), "\n", jsxs(_components.span, {
                    className: "line focused",
                    children: [jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "    </"
                    }), jsx(_components.span, {
                      style: {
                        color: "#005CC5",
                        "--shiki-dark": "#8DDB8C"
                      },
                      children: "Providers"
                    }), jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "> "
                    })]
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    children: jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "  );"
                    })
                  }), "\n", jsx(_components.span, {
                    className: "line",
                    children: jsx(_components.span, {
                      style: {
                        color: "#24292E",
                        "--shiki-dark": "#ADBAC7"
                      },
                      children: "}"
                    })
                  })]
                })
              })
            })
          })]
        })]
      }), jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "use-the-sdk",
          children: ["Use the SDK", jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#use-the-sdk",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsx(_components.p, {
          children: "To use a specific kit, simply import and utilize its components. Whisk SDK is optimized for tree-shaking, so only the imports you use will be included in your bundle."
        }), jsx(_components.p, {
          children: "For example, the Identity Kit:"
        }), jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed has-focused",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            children: jsxs(_components.code, {
              children: [jsxs(_components.span, {
                className: "line focused",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "import"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " { Name, Avatar } "
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "from"
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: ' "@paperclip-labs/whisk-sdk/identity"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "; "
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "function"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#DCBDFB"
                  },
                  children: " ExampleComponent"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "() {"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#6A737D",
                    "--shiki-dark": "#768390"
                  },
                  children: "    // This could come from a connected wallet via Wagmi, ethers.js, web3.js, Privy, etc."
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "    const"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " address"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: " ="
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: ' "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ";"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                "data-empty-line": true,
                children: " "
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "    return"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " ("
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        <"
                }), jsx(_components.span, {
                  style: {
                    color: "#22863A",
                    "--shiki-dark": "#8DDB8C"
                  },
                  children: "div"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " className"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "="
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"flex items-center gap-2"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ">"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            <"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#8DDB8C"
                  },
                  children: "Avatar"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " address"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#F47067"
                  },
                  children: "{"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "address"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#F47067"
                  },
                  children: "}"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " size"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#F47067"
                  },
                  children: "{"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: "30"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#F47067"
                  },
                  children: "}"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " /> // [!code focus]"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "            <"
                }), jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#8DDB8C"
                  },
                  children: "Name"
                }), jsx(_components.span, {
                  style: {
                    color: "#6F42C1",
                    "--shiki-dark": "#6CB6FF"
                  },
                  children: " address"
                }), jsx(_components.span, {
                  style: {
                    color: "#D73A49",
                    "--shiki-dark": "#F47067"
                  },
                  children: "="
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#F47067"
                  },
                  children: "{"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "address"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#F47067"
                  },
                  children: "}"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " /> // [!code focus]"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "        </"
                }), jsx(_components.span, {
                  style: {
                    color: "#22863A",
                    "--shiki-dark": "#8DDB8C"
                  },
                  children: "div"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ">"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    );"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "}"
                })
              })]
            })
          })
        })]
      })]
    }), "\n", jsxs(_components.h2, {
      id: "graphql",
      children: ["GraphQL", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#graphql",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Whisk data is all server via a GraphQL API. The SDK helps abstract this, but if you need direct data access checkout the ", jsx(_components.a, {
        href: "/graphql",
        children: "GraphQL docs"
      }), "."]
    }), "\n", jsxs(_components.h2, {
      id: "customization",
      children: ["Customization", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#customization",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["You can customize all components via configuring the Whisk SDK Theme with CSS variables, or with ", jsx(_components.a, {
        href: "https://tailwindcss.com/",
        children: "tailwindCSS"
      }), " using the className prop."]
    }), "\n", jsx(_components.p, {
      children: "More docs on this coming soon."
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
