// vocs/_lib/vite/.vocs/dist/assets/graphql-CxJntNi_.js
import { jsx, jsxs, Fragment } from "react/jsx-runtime";
import { useMDXComponents } from "vocs/mdx-react";
import { G as GenerateKey } from "./GenerateKey-DHuufDZQ.js";
import "react";
import "lucide-react";
import "clsx";
const frontmatter = {
  "title": "GraphQL",
  "description": "Directly query the Whisk server"
};
function _createMdxContent(props) {
  const _components = {
    a: "a",
    code: "code",
    div: "div",
    h1: "h1",
    h2: "h2",
    h3: "h3",
    header: "header",
    img: "img",
    p: "p",
    pre: "pre",
    span: "span",
    ...useMDXComponents(),
    ...props.components
  };
  return jsxs(Fragment, {
    children: [jsxs(_components.header, {
      children: [jsxs(_components.h1, {
        id: "graphql",
        children: ["GraphQL", jsx(_components.a, {
          "aria-hidden": "true",
          tabIndex: "-1",
          href: "#graphql",
          children: jsx(_components.div, {
            "data-autolink-icon": true
          })
        })]
      }), jsx(_components.div, {
        role: "doc-subtitle",
        children: "Directly query the Whisk server"
      })]
    }), "\n", jsx(_components.p, {
      children: "All data in Whisk is served via a GraphQL API. Whisk SDK's components and hooks abstract this away (but use it under the hood).\nIf you need direct data access or want to use Whisk outside of React, you can use the GraphQL API directly."
    }), "\n", jsxs(_components.p, {
      children: ["API: ", jsx(_components.code, {
        children: "https://api.whisk.so/graphql"
      })]
    }), "\n", jsxs(_components.h2, {
      id: "playground",
      children: ["Playground", jsx(_components.a, {
        "aria-hidden": "true",
        tabIndex: "-1",
        href: "#playground",
        children: jsx(_components.div, {
          "data-autolink-icon": true
        })
      })]
    }), "\n", jsxs(_components.p, {
      children: ["The ", jsx(_components.a, {
        href: "https://api.whisk.so/graphql",
        children: "Whisk GraphQL playground"
      }), " is a great way to explore the API and test queries. To use the playground:"]
    }), "\n", jsxs(_components.div, {
      "data-vocs-steps": "true",
      children: [jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "generate-api-key-if-you-havent-already",
          children: ["Generate API key if you haven't already", jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#generate-api-key-if-you-havent-already",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsx(_components.p, {
          children: "The API key will be sent to your email address. This is the same key you use for the SDK."
        }), jsx(GenerateKey, {})]
      }), jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "add-authorization-header",
          children: ["Add authorization header", jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#add-authorization-header",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsxs(_components.p, {
          children: ["In the ", jsx(_components.code, {
            children: "Headers"
          }), " section of the playground, add the following:"]
        }), jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            "data-title": "Headers",
            "data-lang": "json",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "{"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#005CC5",
                    "--shiki-dark": "#8DDB8C"
                  },
                  children: '  "Authorization"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"Bearer <YOUR_API_KEY>"'
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "}"
                })
              })]
            })
          })
        })]
      }), jsxs(_components.div, {
        "data-depth": "3",
        children: [jsxs(_components.h3, {
          id: "query-the-api",
          children: ["Query the API", jsx(_components.a, {
            "aria-hidden": "true",
            tabIndex: "-1",
            href: "#query-the-api",
            children: jsx(_components.div, {
              "data-autolink-icon": true
            })
          })]
        }), jsx(_components.p, {
          children: "Try running a query like this:"
        }), jsx(Fragment, {
          children: jsx(_components.pre, {
            className: "shiki shiki-themes github-light github-dark-dimmed",
            style: {
              backgroundColor: "#fff",
              "--shiki-dark-bg": "#22272e",
              color: "#24292e",
              "--shiki-dark": "#adbac7"
            },
            tabIndex: "0",
            "data-title": "Example Query",
            "data-lang": "graphql",
            children: jsxs(_components.code, {
              children: [jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "{"
                })
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "  identity"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "("
                }), jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "address"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ": "
                }), jsx(_components.span, {
                  style: {
                    color: "#032F62",
                    "--shiki-dark": "#96D0FF"
                  },
                  children: '"0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045"'
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: ") {"
                })]
              }), "\n", jsxs(_components.span, {
                className: "line",
                children: [jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "    aggregate"
                }), jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: " {"
                })]
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "      name"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "      avatar"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#E36209",
                    "--shiki-dark": "#F69D50"
                  },
                  children: "      bio"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "    }"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "  }"
                })
              }), "\n", jsx(_components.span, {
                className: "line",
                children: jsx(_components.span, {
                  style: {
                    color: "#24292E",
                    "--shiki-dark": "#ADBAC7"
                  },
                  children: "}"
                })
              })]
            })
          })
        })]
      })]
    }), "\n", jsxs(_components.p, {
      children: ["Example of what your playground should look like:\n", jsx(_components.img, {
        src: "/graphql-example.png",
        alt: "GraphQL Example"
      })]
    })]
  });
}
function MDXContent(props = {}) {
  const { wrapper: MDXLayout } = {
    ...useMDXComponents(),
    ...props.components
  };
  return MDXLayout ? jsx(MDXLayout, {
    ...props,
    children: jsx(_createMdxContent, {
      ...props
    })
  }) : _createMdxContent(props);
}
export {
  MDXContent as default,
  frontmatter
};
